modules/portal/portal-impl/src/main/java/com/enonic/xp/portal/impl/url/AttachmentUrlBuilder.java

```diff
@@ -14,17 +15,29 @@ final class AttachmentUrlBuilder
     protected void buildUrl( final StringBuilder url, final Multimap<String, String> params )
     {
         super.buildUrl( url, params );
-        appendPart( url, this.portalRequest.getContentPath().toString() );
-        appendPart( url, "_" );
-        appendPart( url, "attachment" );
 
-        if ( this.params.isDownload() )
+        boolean isSlashAPI = portalRequest.getRawPath().startsWith( "/api/" );
+
+        if ( isSlashAPI )
         {
-            appendPart( url, "download" );
+            appendPart( url, "attachment" );
+            appendPart( url, RepositoryUtils.getContentRepoName( this.portalRequest.getRepositoryId() ) );
+            appendPart( url, this.portalRequest.getBranch().toString() );
+            if ( this.params.isDownload() )
+            {
+                params.put( "download", "" );
```

Comment:
Does it make query string`?download=` ?

Reply:
Sure

Reply:
I was hoping for `?download` which is equivalent of value `null` instead of empty string

Reply:
Unfortunately, we cannot do it another way because `params` will be applied after the base part of the URL is built. We can only add params to the map, at least for now.